### ISSUE TITLE
Issue with Adversarial VAE detection tutorial

### ISSUE THREAD

**Issue by ludoro** [SeldonIO/alibi-detect#105]
Hello everyone. 

I have been trying to run the tutorial https://docs.seldon.io/projects/alibi-detect/en/stable/examples/ad_advvae_mnist.html#

However, I have been running into some issues. I will list them below:

If I try to download the model from the google cloud library, the load_detector(filepath_ad) does not work, even though the same procedure using load_tf_model worked for loading the MNIST model. It seems that the problem lies with namings, because I tried to trace it back in the source code and the naming goes from AdversarialVAE to AdversarialAE, so I think there is a problem in the pickle files maybe?

I then tried to train by myself the adversarial detector, following the code in the link. However, I start getting again some issues. When calling the init method for AdversarialAE (not AdversarialVAE like written in tutorial), I write: 

```
ad = AdversarialVAE(threshold=.5,  # threshold for adversarial score
                        model=model,
                        encoder_net=encoder_net,  # can also pass VAE model instead
                        decoder_net=decoder_net,  # of separate encoder and decoder
                        latent_dim=latent_dim,
                        samples=2,  # nb of samples drawn by VAE
                        beta=0.  # weight on KL-divergence loss term of latent space
                       )

```
However I get an error saying that there is no latent_dim keyword, and if I remove that line I get an error saying that there is no samples keyword method. Looking at the source code, it seems true:

```
def __init__(self,
                 threshold: float = None,
                 ae: tf.keras.Model = None,
                 model: tf.keras.Model = None,
                 encoder_net: tf.keras.Sequential = None,
                 decoder_net: tf.keras.Sequential = None,
                 model_hl: List[tf.keras.Model] = None,
                 hidden_layer_kld: dict = None,
                 w_model_hl: list = None,
                 temperature: float = 1.,
                 data_type: str = None
                 ) -> None:
```

If I remove all the missing keywords, calling the fit method results in dimension error:
`tensorflow.python.framework.errors_impl.InvalidArgumentError: Matrix size-incompatible: In[0]: [2048,512], In[1]: [50,1568] [Op:MatMul] name: ae/decoder/sequential_1/dense/Tensordot/MatMul/`

Could someone look into this?

**Comment by arnaudvl** [comment]
Hi @ludoro . We released a new version of the library which made some breaking changes for the adversarial component as we upgraded it to the version described in our paper [Adversarial Detection and Correction by Matching Prediction Distributions](https://arxiv.org/abs/2002.09364). The stable docs hadn't updated yet though so still showed the old component which is now removed. It should be ok now: https://docs.seldon.io/projects/alibi-detect/en/stable/examples/ad_ae_cifar10.html.
